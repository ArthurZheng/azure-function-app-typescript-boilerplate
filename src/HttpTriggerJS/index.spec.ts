import HttpTriggerJS from '.';

test('HttpTriggerJS should return 400 when `name` not provided', () => {
  let context = { log: jest.fn(), done: jest.fn() };
  let request = { query: {} };
  HttpTriggerJS(context, request);

  expect(context).toMatchSnapshot();
  expect(context.done).toHaveBeenCalledTimes(1);
});

test('HttpTriggerJS should accept `name` in query string', () => {
  let context = { log: jest.fn(), done: jest.fn() };
  let request = { query: { name: 'Kennie' } };
  HttpTriggerJS(context, request);

  expect(context).toMatchSnapshot();
  expect(context.done).toHaveBeenCalledTimes(1);
});

test('HttpTriggerJS should accept `name` in body', () => {
  let context = { log: jest.fn(), done: jest.fn() };
  let request = { query: {}, body: { name: 'Kennie' } };
  HttpTriggerJS(context, request);

  expect(context).toMatchSnapshot();
  expect(context.done).toHaveBeenCalledTimes(1);
});
