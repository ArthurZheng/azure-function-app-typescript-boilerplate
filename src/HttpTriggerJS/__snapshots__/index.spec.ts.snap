// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HttpTriggerJS should accept \`name\` in body 1`] = `
Object {
  "done": [MockFunction] {
    "calls": Array [
      Array [],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": undefined,
      },
    ],
  },
  "log": [MockFunction] {
    "calls": Array [
      Array [
        "JavaScript HTTP trigger function processed a request.",
      ],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": undefined,
      },
    ],
  },
  "res": Object {
    "body": "Hello Kennie",
  },
}
`;

exports[`HttpTriggerJS should accept \`name\` in query string 1`] = `
Object {
  "done": [MockFunction] {
    "calls": Array [
      Array [],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": undefined,
      },
    ],
  },
  "log": [MockFunction] {
    "calls": Array [
      Array [
        "JavaScript HTTP trigger function processed a request.",
      ],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": undefined,
      },
    ],
  },
  "res": Object {
    "body": "Hello Kennie",
  },
}
`;

exports[`HttpTriggerJS should return 400 when \`name\` not provided 1`] = `
Object {
  "done": [MockFunction] {
    "calls": Array [
      Array [],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": undefined,
      },
    ],
  },
  "log": [MockFunction] {
    "calls": Array [
      Array [
        "JavaScript HTTP trigger function processed a request.",
      ],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": undefined,
      },
    ],
  },
  "res": Object {
    "body": "Please pass a name on the query string or in the request body",
    "status": 400,
  },
}
`;
